tv_apps/smart-system/test/marionette/app_install_dialog_test.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+var Keys = {
+  'enter': '\ue006',
+  'left': '\ue012',
+  'right': '\ue014'
+};
+
+var url = require('url');
+var assert = require('chai').assert;
+var AppInstaller = require('./lib/app_install');
+var Server = require('../../../../shared/test/integration/server');
+
+marionette('app install manager tests', function() {
+
+  var opts = {
+    apps: {},
+    hostOptions: {
+      screen: {
+        width: 1920,
+        height: 1080
+      }
+    }
+  };
+
+  var client = marionette.client({ profile: opts });
+  client.plugin('logger', require('marionette-js-logger'));
+  var system;
+  var appInstaller;
+  var server;
+  var serverManifestURL;
+
+  suiteSetup(function(done) {
+    Server.create(__dirname + '/../apps/hosted-app', function(err, _server) {
+      server = _server;
+      serverManifestURL = server.url('manifest.webapp');
+      done(err);
+    });
+  });
+
+  suiteTeardown(function() {
+    server.stop();
+  });
+
+  setup(function() {
+    system = client.loader.getAppClass('smart-system', 'system', 'tv_apps');
+    appInstaller = new AppInstaller(client);
+    client.helper.waitForElementToDisappear('#os-logo');
```

Comment:
One more thing, only `install app and cancel it` will have intermittent error if this line removed, so weird.